File: backend/Models/Enums.cs
namespace Archiboard.Api.Models;

public enum Cardinality
{
    OneToOne,      // 1..1
    OneToMany,     // 1..*
    ManyToOne,     // *..1
    ZeroToOne,     // 0..1
    OneToZero      // 1..0
}

public enum ApiType
{
    REST,
    SOAP,
    GRPC,
    GraphQL,
    NotDefined
}

public enum SpecType
{
    OpenAPI,
    WSDL
}

public enum Exposure
{
    Public,
    Private
}

public enum BrokerType
{
    Kafka,
    RabbitMQ
}

public enum MessageFormat
{
    Avro,
    ProtoBuf,
    JSON
}

public enum MessageType
{
    Event,
    Command
}

public enum SoftwareComponentType
{
    Library,
    Framework
}

public enum Language
{
    CSharp,        // C#
    Java,
    Typescript,
    Javascript,
    Python,
    GoLang
}

public enum DatabaseTechnology
{
    MySQL,
    PostgreSQL,
    SQLServer,
    MongoDB,
    SQLite,
    Neo4J
}

File: frontend/src/app/models/enums.ts
export enum Cardinality {
  OneToOne = 0,
  OneToMany = 1,
  ManyToOne = 2,
  ZeroToOne = 3,
  OneToZero = 4
}

export enum ApiType {
  REST = 0,
  SOAP = 1,
  GRPC = 2,
  GraphQL = 3,
  NotDefined = 4
}

export enum SpecType {
  OpenAPI = 0,
  WSDL = 1
}

export enum Exposure {
  Public = 0,
  Private = 1
}

export enum BrokerType {
  Kafka = 0,
  RabbitMQ = 1
}

export enum MessageFormat {
  Avro = 0,
  ProtoBuf = 1,
  JSON = 2
}

export enum MessageType {
  Event = 0,
  Command = 1
}

export enum SoftwareComponentType {
  Library = 0,
  Framework = 1
}

export enum Language {
  CSharp = 0,
  Java = 1,
  Typescript = 2,
  Javascript = 3,
  Python = 4,
  GoLang = 5
}

export enum DatabaseTechnology {
  MySQL = 0,
  PostgreSQL = 1,
  SQLServer = 2,
  MongoDB = 3,
  SQLite = 4,
  Neo4J = 5
}

